spec/consensus/light-client.md

```diff
@@ -316,3 +316,70 @@ func Backwards(h1,h2) bool {
   return (hash(h2) == old.Header.hash)
  }
 ```
+
+### Failure Detection Module
+
+The light client must be able to detect failures in the trust assumptions.
+Ideally, it should never trust a header, which was forged by malicious
+validators.
+
+### Connectivity requirement
+
+To meet the above requirement, it must be connected to at least one honest
+full node. This is sufficient so long as there's no fork on the main chain.
+
+If there is a fork on the main chain, it means that two full nodes have decided
+on two different headers for the same height. Then the requirement would at
+least require that the lite client is **connected to one honest full node on
+each branch of the fork**.
+
+_Assumption_: no more than `2f` of the nodes are malicious (`f+1` are honest).
+otherwise, they could create more forks.
+
+In practice, this means connecting to 1 or more geographically distributed full
+nodes (called witnesses), which belong to different companies. Note this number
+does not include the primary full node (called primary), which is used for
+obtaining new headers.
+
+_Remark_: we can't guarantee all witnesses won't follow the same branch of a
+fork. to guarantee that, we'll need the light client to be connected to +1/3 of
+nodes, which is impractical.
+
+### Evidence detection
+
+After the light client verifies a new header (`H`) it received from primary, it
+should cross-check `H` with the headers from all witnesses. Cross-checking
+means comparing hashes of the headers. If any two hashes (or more) diverge,
+there's a fork (on the main chain OR phantom fork targeting this light client).
+
+The light client will then need to validate the header it got from a witness
+(`H1`) and verify it's signed by at least one trusted validator. If either of
```

Comment:
Should we require +1/3 (i.e. CheckSupport returning true) here? Or even if one validator from currently trusted set signed, we should check & send evidence?

Reply:
I guess one is enough. But I will think some more about that...

Reply:
How could one be enough? I thought for sure we'd need +1/3 here (ie. "at least one correct") ...

Reply:
https://github.com/tendermint/spec/pull/71#discussion_r374213456

Reply:
```
Therefore we define a notion of slashable fork which is informally defined as a fork in which bonded validators misbehaved (so they can be punished). To define this notion more precisely, we assume that there exists a main chain C (sequence of blocks), and assume that there is a set of bonded validators BD(t) that consists of all bonded (slashable) processes (current and old validators) at time t. Then a slashable fork at some time t corresponds to the existence of a commit for some block B at height h (B.time < t) that is different from a block at height h of the chain C; furthermore, a commit for the block B contains at least a single valid signature from a process that is part of the set BD(t). Note that detecting a slashable fork requires access to the main chain; therefore classifying if fork is slashable can be done only at the correct full node. On the other hand a lite client can only detect a fork, and then delegate the task of determining if the fork is slashable or not to some correct full node. 
```
from fork detection doc